Backpack/CardButton/Classes/ShareCardButton.swift

```diff
@@ -0,0 +1,114 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UIKit
+
+public class BPKShareCardButton: UIButton, CardButtonProtocol {
+
+    public var style: BPKCardButtonStyle {
+        didSet { updateLookAndFeel() }
+    }
+
+    public var size: BPKCardButtonSize {
+        didSet { updateLookAndFeel() }
+    }
+
+    let containedBackgroundCircle = UIView()
+
+    public init(
+        accessibilityLabel: String,
+        style: BPKCardButtonStyle = .default,
+        size: BPKCardButtonSize = .default
+    ) {
+        self.style = style
+        self.size = size
+
+        super.init(frame: .zero)
+        self.accessibilityLabel = accessibilityLabel
+
+        setup()
+        updateLookAndFeel()
+    }
+
+    @available(*, unavailable)
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func updateLookAndFeel() {
+        updateButtonImages()
+        updateBackgroundCircleView()
+    }
+
+    private func updateButtonImages() {
+        switch size {
+        case .default:
+            let largeIconNormal = BPKIcon.makeLargeIcon(
+                name: .shareiOS,
+                color: normalIconColor(style: style)
+            )
+            setImage(largeIconNormal, for: .normal)
+
+            let largeIconHighlighted = BPKIcon.makeLargeIcon(
+                name: .shareiOS,
+                color: highlightedIconColor(style: style)
+            )
+            setImage(largeIconHighlighted, for: .highlighted)
+        case .small:
+            let smallIconNormal = BPKIcon.makeSmallIcon(
+                name: .shareiOS,
+                color: normalIconColor(style: style)
+            )
+            setImage(smallIconNormal, for: .normal)
+
+            let smallIconHighlighted = BPKIcon.makeSmallIcon(
+                name: .shareiOS,
+                color: highlightedIconColor(style: style)
+            )
+            setImage(smallIconHighlighted, for: .highlighted)
+        }
+    }
+
+    private func normalIconColor(style: BPKCardButtonStyle) -> UIColor {
+        switch style {
+        case .onDark:
+            return BPKColor.textOnDarkColor
+        case .contained:
+            return BPKColor.textPrimaryColor
+        case .default:
+            return BPKColor.textPrimaryColor
+        }
+    }
+
+    private func highlightedIconColor(style: BPKCardButtonStyle) -> UIColor {
+        switch style {
+        case .onDark:
+            return BPKColor.textDisabledOnDarkColor
+        case .contained:
+            return BPKColor.textLinkColor
+        case .default:
+            return BPKColor.textLinkColor
+        }
```

Comment:
This could be good candidates to move the the BPKCardButtonStyle enum, the caller's code cleans up quite a bit, it's quite swifty!
it could even return the BPKIcon instead.
```swift
enum Size {
  func normalIcon(_ style: BPKCardButtonStyle) -> UIImage {
     switch self {
         case .small: BPKIcon.makeSmallIcon(name: .shareiOS, color: style.iconColor) // <- notice the style 'iconColor'
     }
  }
}
//and call it like
private func updateButtonImages() {
  setImage(size.normalIcon(style), for: .normal) 
  setImage(size.highlightedIcon(style), for: .normal)
}
```

Reply:
Totally! what I mention could be implemented as a fileprivate extension of the Size and Style in the Share or Save button class
```swift
class BPKShareButton:... {}
fileprivate extension BPKCardButtonSize {
  func normalIcon()...
}
```